src/filesystem.cpp

```diff
@@ -309,44 +310,47 @@ AsyncIOContext *FileAccess::asyncfopen(string *f)
 
 bool FileAccess::asyncopenf()
 {
+    numAsyncReads++;
```

Comment:
I'd increment this counter only when we're sure the file has been opened. If this function returns `false`, you probably don't want the counter to be incremented. Or maybe I'm wrong and it's possible to have _async reads_ with a file that is not opened?